### PR TITLE
Add failing test on JSON escaping issue

### PR DESCRIPTION
**Pull request by Spy-Seth**
This test is here to discuss about the change made by the commit https://github.com/algolia/react-element-to-jsx-string/commit/5c0ae3ef718259e23510106bc4d0093dd88085d5 to solve the issue #80 on string escaping. The proposed fix introduce a regression on a the JSON representation that is no more correctly escaped.

@vvo I would have your thought about this one, I will propose a PR to correctly fix both cases

### CODE REVIEW

**Comment by vvo** [comment]
Hello, can you provide more context/use case/examples on why the current library state is not good? Sorry there have been a lot of small fixes to that output recently, it's hard to follow.

Phrase it as if I was not the maintainer :D

**Comment by Spy-Seth** [comment]
Lately there have been a lot of change on the simple/double quote representation. The aim of all this modification is to always generate a valid output (correctly escaped).

The test that I've added show a last issue with the escaping:
screenshot 2017-06-28 17 25 41

After some digging the nested string in object props have simple quote as delimiter instead of double quote. All the string escaping system is made to generate string valid in double quote not in single quote. So actually single quote are not escaped witch led to erroneous string representation `'foo'bar'`. 

A first solution is to force all the delimiter to be double quote: `"foo'bar"`. This could be done by adding the options `{ singleQuotes: false }` to the `stringify` call.

A second solution is to keep the revert the https://github.com/algolia/react-element-to-jsx-string/commit/5c0ae3ef718259e23510106bc4d0093dd88085d5 commit who introduce the current issue. Previous this commit simple quote where correctly escaped. But we will to find another solution for quote in case of array representation: `[\'foo\']`

**Comment by vvo** [comment]
I think this module should use double quotes by default for arrays and strings, that would solve this issue right?

Is it what you mean by using singleQuotes: false?

If so, let's do this.

**Comment by vvo** [comment]
I think this module should use double quotes by default for arrays and strings, that would solve this issue right?

Is it what you mean by using singleQuotes: false?

If so, let's do this.

**Comment by Spy-Seth** [comment]
Fixed by https://github.com/algolia/react-element-to-jsx-string/pull/107
